File: CSharpFundamentals/13 ListsAndMatrices/BombNumbers/BombNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BombNumbers
{
    class BombNumbers
    {
        static void Main(string[] args)
        {
            var nums = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();
            long[] bombNums = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();
            long bombNum = bombNums[0];
            long left = bombNums[1];
            long right = bombNums[1];
           
            for (long i = 0; i < nums.Length; i++)
            {             
                if (nums[i] == bombNum)
                {
                    long start = i - left;
                    long end = i + right;
                    if (end > nums.Length - 1)
                    {
                        right = nums.Length - i - 1;
                    }
                    if (start < 0)
                    {
                        start = 0;
                        for (long j = start; j <= i+right; j++)
                        {
                            nums[j] = 0;
                        }
                        //nums.RemoveRange(start, (i-start) + right + 1);
                    }
                    else
                    {
                        for (long j = start; j <= i + right; j++)
                        {
                            nums[j] = 0;
                        }
                       // nums.RemoveRange(start, left + right + 1);
                    }    
                }
            }
            Console.WriteLine(nums.Sum());
        }
    }
}

File: SoftwareTechnologies/Java/JavaBasics/src/BombNumbers.java
import java.util.*;

public class BombNumbers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int[] nums = Arrays.stream(
                scan.nextLine().split(" "))
                .mapToInt(Integer::parseInt).toArray();

        int bombNum = scan.nextInt();
        int left = scan.nextInt();
        int right = left;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == bombNum) {
                int start = i - left;
                int end = i + right;
                if (end > nums.length - 1) {
                    right = nums.length - i - 1;
                }
                if (start < 0) {
                    start = 0;
                    for (int j = start; j <= i + right; j++) {
                        nums[j] = 0;
                    }
                }
                else {
                    for (int j = start; j <= i + right; j++) {
                        nums[j] = 0;
                    }
                }
            }
        }
        long sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum = sum + nums[i];
        }
        System.out.println(sum);
    }
}
